CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/server/RequestHandler/ReplicatorConfigurationRequestHandler.java

```diff
@@ -327,4 +362,182 @@ public boolean filtered(Document document, EnumSet<DocumentFlag> flags) {
+        String remoteDocId = remoteDoc.getId();
+        String localDocId = localDoc.getId();
+        if (remoteDocId != localDocId) {
+            Log.e(TAG, "Remote docId and local docId are different");
+        }
+        if (remoteDocId != docId) {
+            Log.e(TAG, "Remote docId doesn't match with conflict docId");
+        }
+        if (docId != localDocId) {
+            Log.e(TAG, "Local docId doesn't match with conflict docId");
+        }
+        MutableDocument newDoc = localDoc.toMutable();
+        Map<String, Object> remoteDocMap = remoteDoc.toMap();
+        for (Map.Entry<String, Object> entry : remoteDocMap.entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+            if (!newDoc.contains(key)) {
+                newDoc.setValue(key, value);
+            }
+        }
+        return newDoc;
+    }
+}
+
+class IncorrectDocIdConflictResolver implements ConflictResolver {
+    private static final String TAG = "CCRREPLCONFIGHANDLER";
+    @Override
+    public Document resolve(Conflict conflict) {
+        Document localDoc = conflict.getLocalDocument();
+        Document remoteDoc = conflict.getRemoteDocument();
+        String docId = conflict.getDocumentId();
+        String remoteDocId = remoteDoc.getId();
+        String localDocId = localDoc.getId();
+        if (remoteDocId != localDocId) {
+            Log.e(TAG, "Remote docId and local docId are different");
+        }
+        if (remoteDocId != docId) {
+            Log.e(TAG, "Remote docId doesn't match with conflict docId");
+        }
+        if (docId != localDocId) {
+            Log.e(TAG, "Local docId doesn't match with conflict docId");
+        }
+        String newId = "changed" + docId;
+        MutableDocument newDoc = new MutableDocument(newId, localDoc.toMap());
+        newDoc.setValue("new_value", "couchbase");
+        return newDoc;
+    }
+}
+
+class DelayedLocalWinConflictResolver implements ConflictResolver {
+    private static final String TAG = "CCRREPLCONFIGHANDLER";
+    @Override
+    public Document resolve(Conflict conflict) {
+        Document localDoc = conflict.getLocalDocument();
+        Document remoteDoc = conflict.getRemoteDocument();
+        String docId = conflict.getDocumentId();
+        String remoteDocId = remoteDoc.getId();
+        String localDocId = localDoc.getId();
+        if (remoteDocId != localDocId) {
+            Log.e(TAG, "Remote docId and local docId are different");
+        }
+        if (remoteDocId != docId) {
+            Log.e(TAG, "Remote docId doesn't match with conflict docId");
+        }
+        if (docId != localDocId) {
+            Log.e(TAG, "Local docId doesn't match with conflict docId");
+        }
+        try {
+            sleep(1000 * 10);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        return localDoc;
+    }
+}
+
+class DeleteDocConflictResolver implements ConflictResolver {
```

Comment:
what the purpose of the DeleteDocConflictResolver class is? the question is for all platforms

Reply:
as the case name suggested, we are overwriting delete operation with local doc copy. Doc is delete on remote end, but we want to retain the doc anyways.